src/index.js

```diff
@@ -1,43 +1,136 @@
+// task Class: represents a task object
+class Task {
+  constructor(id, description, completed) {
+    this.id = id;
+    this.description = description;
+    this.completed = completed;
   }
-};
+}
+// Store Class: Handles Storage
 
-createList(tasks);
+// Creating new Store instance
+const store = new Store();
+
+// Update Task
+function updateTask(e) {
+  const text = e.target.value;
+  const value = e.target.id;
+  const splitid = value.split('-');
+  const idstring = splitid[2];
+  const id = parseInt(idstring, 10);
+  const list = store.getList();
+  for (let i = 0; i < list.length; i += 1) {
+    const task = list[i];
+    if (task.id === id) {
+      list[i].description = text;
+    }
+  }
+  localStorage.setItem('list', JSON.stringify(list));
+}
+
+const inputTasks = [];
+// UI Class : Handles UI tasks
+class UI {
+  // Static so I don't have to instantiate
+
+  static displayList() {
+    const list = store.getList();
+    list.forEach((task) => UI.addTodoList(task));
+  }
+
+  static addTodoList(task) {
+    const tasksList = document.getElementById('list-container');
+
+    const taskContent = document.createElement('li');
+    taskContent.innerHTML = `
+        <input type="checkbox" id="checkbox" class="checkbox" />
+        <div><input value="${task.description}" class="task-input" id="task-input-${task.id}"/></div>
+        <div class="dots-container"><img class="dots"src="${Dots}" /></div>
+    `;
+    tasksList.appendChild(taskContent);
+    const taskInput = taskContent.querySelector('.task-input');
+    taskInput.addEventListener('keyup', updateTask);
+    inputTasks.push(taskInput);
+    taskContent.classList.add('element');
+    taskContent.setAttribute('id', `task-${task.id}`);
+  }
+
+  static deleteCompleted() {
+    const liList = document.querySelectorAll('.element');
+    liList.forEach((li) => {
+      const checkbox = li.querySelector('.checkbox');
+      if (checkbox.checked) {
+        li.remove();
+        const list = localStorage.getItem('list');
+        const parsedlist = JSON.parse(list);
+        //
+        const filteredList = parsedlist.filter((task) => {
+          const fullid = li.id;
+
+          const idString = fullid.split('-')[1];
+          const id = parseInt(idString, 10);
+          return task.id !== id;
+        });
+        localStorage.setItem('list', JSON.stringify(filteredList));
+        store.resetIds();
+      }
+    });
+  }
+
+  static clearFields() {
+    document.querySelector('#list-item').value = '';
+  }
+}
+
+// Update Task
+
+// Event listener for UI when load
+document.addEventListener('DOMContentLoaded', UI.displayList);
+
+const btnClear = document.querySelector('#btn-clear');
+btnClear.addEventListener('click', UI.deleteCompleted);
+
+// Event listener for submit
+document.querySelector('#new-todo-form').addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  const id = store.count;
+  const description = document.querySelector('#list-item').value;
```

Comment:
[Optional] Currently, the app allows me to enter nothing or only empty spaces. I suggest that you trim the value of the input before saving in order to avoid this.

![image](https://user-images.githubusercontent.com/20554583/200082846-b7cfa1a9-47f2-4db7-991d-6cb5144074e5.png)